composer/loggers/neptune_logger.py

```diff
@@ -299,6 +303,16 @@ def post_close(self) -> None:
             self._neptune_run.stop()
             self._neptune_run = None
 
+        if self._oom_observer:
+            self._log_oom_snapshots()
+
+    def _log_oom_snapshots(self) -> None:
+        if self._oom_observer is None or self._oom_observer.filename_config is None:
+            return
+
+        for file_name in self._oom_observer.filename_config.list_filenames():
+            self.base_handler[f'{NeptuneLogger.oom_snaphot_namespace}/{file_name}'].upload(file_name)
+
```

Comment:
I'm wondering if this is the correct way the callbacks should be connected to the loggers. I was under the impression the loggers were agnostics to the callbacks, and the callbacks would be the one pushing the artifacts. In this specific case, the OOMObserver should push the files when they are generated or on `close`. All loggers would benefit from capturing the files associated to the OOM snapshot, even if they can't render it.

Reply:
This is a good point -- sorry I missed this in review and rushed this.

OOMObserver does log to callbacks:
```
logger.upload_file(remote_file_name=remote_file_name, file_path=f, overwrite=self.overwrite)
```

@AleksanderWWW do you know why this is needed! 

Thank you for flagging @antoinebrl  :)

Reply:
Hey,
apologies, I might have misunderstood how the callback works. Next week I will open a PR that will address those changes more carefully. We will remove this way of uploading OOM files and instead work on our normal `upload_file` functionality to handle the oom traces better.
Once again, apologies for the confusion.

Reply:
Thanks for pointing this out, @antoinebrl ! 🙌 

Since @AleksanderWWW had consulted me internally before creating this PR, I am at least partly to blame for the confusion 🙈 

We'll create a new PR to address these and a few other minor doc changes.

@mvpatel2000 - when is your next release scheduled?

Reply:
We plan to cut next Tuesday and release by Thursday. Is it possible to get this in before? I'm happy to help out. Otherwise, we could also revert this and do a follow-on PR with changes.

We plan to release more frequently (every 2-3 weeks) so it wouldn't be a huge deal to punt

Reply:
I'd prefer reverting and creating a new PR if we cannot meet the Tuesday cutoff. Don't want to expose users to this change :)

@AleksanderWWW - wdyt? Can we meet the Tuesday cutoff?

Reply:
Ok, I will revert for now.